File: Uitwerkingen_Boek/progs/h3o5.py
import cmath

N = 64
x = [0 for i in range(N)]
for n in range(N):
    x[n] = 1 + cmath.cos(n * cmath.pi / 32) + cmath.sin(n * cmath.pi / 4)
for k in range(N):
    sum = 0
    for n in range(N):
        sum += x[n] * cmath.exp(-1j * 2 * cmath.pi * k * n / N)
    a_k = sum/N
    if (abs(a_k.real) < 1e-9 and abs(a_k.imag) < 1e-9):
        a_k = 0
    print('|a_' + str(k) + '| = {0:.4f}'.format(abs(a_k)), 'arg(a_'+str(k)+') = {0:.4f} pi'.format(cmath.phase(a_k)/cmath.pi))

File: Uitwerkingen_Boek/progs/h3o5.m
N = 64;
n = [0:1:N-1];
x_n = 1 + cos(n*pi/32) + sin(n*pi/4);
a_k = fft(x_n)/N;
for i = 1:length(a_k)
    if abs(real(a_k(i)))<1e-9 && abs(imag(a_k(i)))<1e-9
        a_k(i) = 0;
    end
    fprintf("a_%d = %.4f%+.4fj, |a_%d| = %.4f, arg(a_%d) = %.4f pi\n", i-1, real(a_k(i)), imag(a_k(i)), i-1, abs(a_k(i)), i-1, angle(a_k(i))/pi)
end
